enhancements/network/ovn-hypershift-live-migration.md

```diff
@@ -0,0 +1,295 @@
+
+There are some network requirements, some of them are imposed by kubevirt 
+live migration and others by kubernetes:
+
+- Consistent IP follows VM during Live Migration, and is the IP of the primary guest node interface that the kubelet uses to register/communicate with.
+- Guest Cluster Network isolation. Guest clusters shouldn’t be able to talk to other guest clusters directly except through public LBs. Guest clusters shouldn’t be able to talk to any infra components except through public LBs
+- No TCP connection breakage over E/W and N/S during live migration with minimal disruption.
+- Support exposing workers with services 
+- Allow workers services acess to services like kubedns and apiserver from hosted control plane.
+
+To accomplish that the ovn-kubernetes default network handling has to implement some features triggered when a 
+hypershift kubevirt VM is detected.
+
+## Motivation
+
+At kubevirt every VM is backed up by a "virt-launcher" pod; how the "virt-launcher" pod net interface is "bind" to the VM is specified by the network binding at the VM spec.
+
+The main VM network binding mechanism are:
+- masquerade binding: the VM has a stable fake address that is masqueraded (SNAT) at virt-launcher pod all the VMs have the same IP.
+- bridge binding: The virt-launcher interface is replaced by a bridge with one leg on the pod interface and the other at the VM tap device, a DHCP server is started with the ip from the pod so DHCP aware VMs get it.
+
+Since Masquerade binding has a fake address at VM guest it cannot be used for hypershift, the kubernetes nodes should see the address they are using to register to the api server.
+
+The bridge binding is able to expose the pod ip to the VM as expected by hypershift but kubevirt does not support live migration with it, for kubevirt to support live migration with bridge binding 
+at primary interface the OVN kubernetes CNI need to do the following:
+- not set the IP at pod netns
+- not expect the IP to be on the pod netns
+
+### Goals
+
+Hypershift kubevirt provider needs the live-migration functionality, missing features at CNI need to be implement at the ovn-kuberentes 
+which is the default CNI for openshift.
+
+### Non-Goals
+
+- Support kubevirt live-migration out of hypershift context
+
+## Proposal
+
+The gist of the design is:
+
+1. Hypershift kubevirt workers attach to ovn-k node switches but CNI does not set address at virt-launcher pods netns
+2. DHCP is configured at VMs logical switch ports to deliver the address to VMs
+3. Routing is done using /32 address so one node's subnet ip can be routed from different node
+3. The VM's gateway ip and mac are independent of the node they are running on using proxy arp
+4. The [proxy arp](https://github.com/dceara/ovn/tree/bz2155306-arp-proxy-poc) functionality need to be implemented at ovn 
```

Comment:
would it make sense to elaborate a bit on this solution ? Maybe an annex at the end of the doc.

Reply:
In fact this story has changed, we discover that we have already a basic proxy arp feature at ovn but we need to add some features.

Will reprhase it.